app/src/renderer/vuex/modules/governance/proposals.js

```diff
@@ -70,22 +66,12 @@ export default ({ node }) => {
     async getProposal({ state, commit }, proposal_id) {
       state.loading = true
       try {
-        let final_tally_result
         state.error = null
         state.loading = false
         state.loaded = true // TODO make state for single proposal
         const proposal = await node.getProposal(proposal_id)
-        commit(`setProposal`, proposal.value)
-        if (proposal.value.proposal_status === `VotingPeriod`) {
-          final_tally_result = await node.getProposalTally(
-            proposal.value.proposal_id
-          )
-        } else {
-          final_tally_result = JSON.parse(
-            JSON.stringify(proposal.value.final_tally_result)
-          )
-        }
-        commit(`setProposalTally`, { proposal_id, final_tally_result })
+        setProposalTally(commit, node)(proposal)
+        return proposal
       } catch (error) {
         commit(`notifyError`, {
```

Comment:
please remove the notify and add loading and loaded properties in the catch block

Reply:
Are you sure? I can't find any other place where we do that

Reply:
I will move the flag switch after the real action, as it is now it doesn't make sense, I haven't seen this earlier

Reply:
I think I will never see the error, we had a little discussion with @faboweb about this, I'll open an issue where we can develop

Here the loaded status is shared across `getProposals` and `getProposal`, when you access a proposal that `proposalId` is going to the `proposals` dictionary, but loading, loaded and error are promiscuous, same happen for transactions etc.. 

We were thinking of changing the priority in TmPage so error rule, and error in this case should not come from `proposals.error` but from either:
- `proposals.statuses[proposalId]` if we want a structure with `{loading, loaded, error}`
- `proposals.errors[proposalId]` if we want to store just errors

All this apply only to proposals that got accessed directly from URL, the others won't mind, they are already in the store thanks to `getProposals` and adding a bunch of `true false null` for every proposal scanned is not really useful I think, so we should expect those flags to be also undefined in case of succesfully fetched proposals

This shuold be generic so we can replace the word `proposal` with transaction etc..